NAMESPACE

```diff
@@ -1,19 +1,29 @@
 # Generated by roxygen2: do not edit by hand
 
 export("%>%")
+export(h1_tabstop)
+export(h2_tabstop)
+export(h3_tabstop)
+export(h4_tabstop)
+export(h5_tabstop)
+export(h6_tabstop)
+export(nhs_card)
+export(nhs_card_tabstop)
+export(nhs_footer)
+export(nhs_grid_2_col)
+export(nhs_grid_3_col)
+export(nhs_grid_4_col)
+export(nhs_header)
+export(nhs_navlistPanel)
+export(nhs_selectInput)
+export(nhs_selectizeInput)
 export(run_app)
-export(theme_nhsbsa)
```

Comment:
Can you remove and edit it? I never touch this as it says don't edit. how did you add all these exports? did you add by manually? (just curious)

Reply:
Nothing in this is edited by hand. `roxygen2` package handles everything from instructions given in documentation comments. I'll add an example in a sec.

Reply:
So if you look on L29 of this file, there is the entry `importFrom(rlang,.data)`. This is added because `roxygen2` when the `roxygenise()` function is called (i.e. package is documented) sees L8 in the file `R/nhsbsaShinyR.R` which is `@importFrom rlang := .data`.

Similarly you can import an entire package, as is done for `shiny` by including `@import shiny` in at least one documentation comment in any file in the `R` folder. For example, that is in the `R/app_ui.R` twice, and `R/app_server.R` once. The effect would be the same if we deleted `@import shiny` from the ui file entirely.

Reply:
Note that there is no `import nhsbsaR` or `importFrom(nhsbsaR, theme_nhsbsa_highchart)` in `NAMESPACE`. This is because we instead namespace the call to `theme_nhsbsa_highchart` by using `nhsbsaR::theme_nhsbsa_highchart()` in `R/mod_chart_example.R`.

Reply:
| Code| Effect |
|--------|--------|
| `#' import package` | `package` is globally available and any of its exported functions can be called directly, without namespacing with `package::function`. |
| `#' importFrom package func1 func2` | `func1` and `func2` are available globally and can be called without namespacing. But `func3` in `package` would need `package::func3`. |